### PR TITLE
refactor: address file-extensions review comments

### PR DESCRIPTION
**Pull request by jviotti**
See: https://github.com/resin-io/etcher/pull/1343#pullrequestreview-34288142
Signed-off-by: Juan Cruz Viotti <jviotti@openmailbox.org>

### CODE REVIEW

File: lib/gui/models/store.js [hunk 1]
```diff
@@ -344,7 +344,9 @@ const storeReducer = (state = DEFAULT_STATE, action) => {
         });
       }
 
-      if (fileExtensions.getLastFileExtension(action.data.path) !== action.data.extension) {
+      const imageLastExtension = fileExtensions.getLastFileExtension(action.data.path);
```
Comment (lurch) [inline lib/gui/models/store.js]: Only a minor nit, but I think `lastImageExtension` sounds more natural than `imageLastExtension`.

File: tests/shared/file-extensions.spec.js [hunk 2]
```diff
@@ -29,62 +29,66 @@ describe('Shared: fileExtensions', function() {
...
-        extension: [ 'img', 'bz2' ]
+        extensions: [ 'img', 'bz2' ]
       },
       {
         file: 'path/to/filename.img.xz',
-        extension: [ 'img', 'xz' ]
+        extensions: [ 'img', 'xz' ]
+      },
+      {
+        file: 'path/to/filename.img.gz',
```
Comment (lurch) [inline tests/shared/file-extensions.spec.js]: You've now got `'path/to/filename.img.gz'` on line 47 and line 59 ??

File: tests/gui/models/selection-state.spec.js [hunk 3]
```diff
@@ -512,6 +512,40 @@ describe('Browser: SelectionState', function() {
...
+              archiveExtension: 'xz',
+              size: {
+                original: 999999999,
+                final: {
+                  estimation: false,
+                  value: 999999999
+                }
+              }
+            });
+          }).to.throw('Invalid image archive extension: foo');
```
Comment (lurch) [inline tests/gui/models/selection-state.spec.js]: Huh?

Comment (lurch) [inline tests/gui/models/selection-state.spec.js]: Although from the look of the CI tests, it seems this isn't failing as expected?!?

File: lib/gui/models/store.js (hunk 1 above)
Comment (jviotti) [inline lib/gui/models/store.js]: Fixed!

File: tests/gui/models/selection-state.spec.js (hunk 3 above)
Comment (jviotti) [inline tests/gui/models/selection-state.spec.js]: Uh, I was testing this out, then got dragged to a meeting, and then though it as complete :P I'll fix it in a second.

File: tests/shared/file-extensions.spec.js (hunk 2 above)
Comment (jviotti) [inline tests/shared/file-extensions.spec.js]: Fixed!

Comment (jviotti) [inline tests/shared/file-extensions.spec.js]: I should seriously stop multi tasking, otherwise I'll keep committing non sense stuff :P

Comment (lurch) [inline tests/shared/file-extensions.spec.js]: Well as long as you're happy for me to keep catching the little slip-ups you make ;-)

File: tests/gui/models/selection-state.spec.js [hunk 4]
```diff
@@ -512,6 +512,40 @@ describe('Browser: SelectionState', function() {
...
+                original: 999999999,
+                final: {
+                  estimation: false,
+                  value: 999999999
+                }
+              }
+            });
+          }).to.throw('Invalid image extension: ifg');
+        });
+
```
Comment (lurch) [inline tests/gui/models/selection-state.spec.js]: From what I can tell of https://github.com/resin-io/etcher/pull/1347/files#diff-74bcbd72e100aecd2f68e833e7a169f4R342 this is a newly-caught error? Bonus!
Could we also add a test for `{ path: 'foo.ifg', extension: 'ifg' }` ?  (and perhaps `{ path: 'foo.ifg.gz', extension: 'ifg', archiveExtension: 'gz' }` too?)

File: tests/shared/file-extensions.spec.js (hunk 2 above)
Comment (jviotti) [inline tests/shared/file-extensions.spec.js]: Haha, sounds good to me! :P

File: tests/gui/models/selection-state.spec.js (hunk 4 above)
Comment (jviotti) [inline tests/gui/models/selection-state.spec.js]: Done!

**Comment by jviotti** [comment]
@lurch I noticed an issue with the wording on this PR. We use `archiveExtension` to also set compressed extensions that are not archives, like `gz`. We could rename `archiveExtension` to `compressedExtension` but again not every "wrapper" format is compressed (think `tar`).

Looks like we need a generic word that means that this is the extension of the format that wraps the image. What do you think?

**Comment by lurch** [comment]
> @lurch I noticed an issue with the wording on this PR. We use archiveExtension to also set compressed extensions that are not archives, like gz. We could rename archiveExtension to compressedExtension but again not every "wrapper" format is compressed (think tar).

Yeah, I noticed that too, but given that this is only for the analytics, I think `archiveExtension` is suitably vague to cover both 'archive' and 'compressed' formats.
Ping @stefan-mihaila and @alexandrosm in case they have any comments.

File: tests/gui/models/selection-state.spec.js [hunk 5]
```diff
@@ -512,6 +512,73 @@ describe('Browser: SelectionState', function() {
...
+              size: {
+                original: 999999999,
+                final: {
+                  estimation: false,
+                  value: 999999999
+                }
+              }
+            });
+          }).to.throw('Invalid image extension: ifg');
+        });
```
Comment (lurch) [inline tests/gui/models/selection-state.spec.js]: Are there any unit-tests (of physical files, and not just filenames) that check that a `foo.ifg` wrapped inside a `foo.zip` "fails as expected" ?

Comment (jviotti) [inline tests/gui/models/selection-state.spec.js]: The redux store is only concerned with the information an image object provides. The image format archive handlers are in charge of setting both `extension` and `archiveExtension`, and that is covered by the `image-stream` tests.

**Comment by jviotti** [comment]
> but given that this is only for the analytics, I think archiveExtension is suitably vague to cover both 'archive' and 'compressed' formats.

I still feel it sounds awkward. What do you think about `wrapperExtension`?

**Comment by lurch** [comment]
I'd be happy with `wrapperExtension`, but maybe it feels a bit clunky? Lets see what the other guys say.

**Comment by jviotti** [comment]
I'll merge this, but happy to send another PR if needed after we agree on the name.

**Pull request merged**